rskj-core/src/test/java/co/rsk/peg/BridgeSupportPegoutCreationTest.java

```diff
@@ -0,0 +1,172 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.config.BridgeConstants;
+import co.rsk.crypto.Keccak256;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import org.apache.commons.lang3.RandomUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.Block;
+import org.ethereum.core.Repository;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BridgeSupportPegoutCreationTest extends BridgeSupportTestBase {
+    @Before
+    public void setUpOnEachTest() {
+        activations = mock(ActivationConfig.ForBlock.class);
+        bridgeSupportBuilder = new BridgeSupportBuilder();
+    }
+
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_before_RSKIP298_activation() {
+        when(activations.isActive(ConsensusRule.RSKIP298)).thenReturn(false);
+
+        Sha256Hash btcTxHash = PegTestUtils.createHash(RandomUtils.nextInt());
+        Keccak256 rskTxHash = PegTestUtils.createHash3(RandomUtils.nextInt());
+
+        BridgeConstants bridgeConstants = this.bridgeConstantsRegtest;
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        when(provider.getPegoutCreationRskTxHashByBtcTxHash(any())).thenReturn(Optional.of(rskTxHash));
+        Repository repository = createRepository();
+        BtcBlockStoreWithCache.Factory mockFactory = mock(BtcBlockStoreWithCache.Factory.class);
+        Block executionBlock = Mockito.mock(Block.class);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withProvider(provider)
+            .withBridgeConstants(bridgeConstants)
+            .withActivations(activations)
+            .withBtcBlockStoreFactory(mockFactory)
+            .withExecutionBlock(executionBlock)
+            .withRepository(repository)
+            .build();
+
+        Optional<Keccak256> pegoutRskTxHash = bridgeSupport.getPegoutCreationRskTxHashByBtcTxHash(btcTxHash);
+
+        assertFalse(pegoutRskTxHash.isPresent());
```

Comment:
This is weird, you mocked the provider above to return `Optional.of(rskTxHash)`. So why is it returning empty? What am I missing?